SteelHead/Azure-Cloud-Cookbooks/101-service-chain-gw-appliance/scripts/Template-Runbook-AccelerationRoutes.ps1

```diff
@@ -85,28 +85,29 @@ Import-Module Az.Accounts
 Import-Module Az.Network
 Import-Module Az.Automation
 
+#Use system-maganed identity instead of RunAsAccount
 try
 {
-    $connectionName = "AzureRunAsConnection"
-    $servicePrincipalConnection=Get-AutomationConnection -Name $connectionName         
-
     "Logging in to Azure..."
-    Add-AzAccount `
-        -ServicePrincipal `
-        -TenantId $servicePrincipalConnection.TenantId `
-        -ApplicationId $servicePrincipalConnection.ApplicationId `
-        -CertificateThumbprint $servicePrincipalConnection.CertificateThumbprint 
+    Connect-AzAccount -Identity
 }
-catch 
-{
-    if (!$servicePrincipalConnection)
+catch {
+    Write-Error -Message $_.Exception
+    throw $_.Exception
+}
+
+#Get all Resource Manager resources from all resource groups

```

Comment:
It is great.
Just lines from 99 to 112, do you agree and can you please update the PR?

Reply:
No it's the whole region block as specified.  Here is the lines from my working code in production:

#region lib

Import-Module Az.Accounts
Import-Module Az.Network
Import-Module Az.Automation

try
{
    "Logging in to Azure..."
    Connect-AzAccount -Identity
}
catch {
    Write-Error -Message $_.Exception
    throw $_.Exception
}

#Get all Resource Manager resources from all resource groups
$ResourceGroups = Get-AzResourceGroup

foreach ($ResourceGroup in $ResourceGroups)
{    
    Write-Output ("Showing resources in resource group " + $ResourceGroup.ResourceGroupName)
    $Resources = Get-AzResource -ResourceGroupName $ResourceGroup.ResourceGroupName
    foreach ($Resource in $Resources)
    {
        Write-Output ($Resource.Name + " of type " +  $Resource.ResourceType)
    }
    Write-Output ("")
}

#endregion

Reply:
I see, so here there is no need to browse all the resources from all resource groups and display name and type, right?
We could just get rid of those lines:

`#Get all Resource Manager resources from all resource groups
$ResourceGroups = Get-AzResourceGroup

foreach ($ResourceGroup in $ResourceGroups)
{
Write-Output ("Showing resources in resource group " + $ResourceGroup.ResourceGroupName)
$Resources = Get-AzResource -ResourceGroupName $ResourceGroup.ResourceGroupName
foreach ($Resource in $Resources)
{
Write-Output ($Resource.Name + " of type " + $Resource.ResourceType)
}
Write-Output ("")
}`

Reply:
Yeah sorry I don't know how that enumeration got in there.  I don't see the variable used in the rest of the code.

Reply:
No problem, thanks for your contribution!